### PR TITLE
1669: Move the duplicated setup code for LB stats into separate function

### LINKED ISSUE #1669: Move the duplicated setup code for `LBStats` into separate function
See https://github.com/DARMA-tasking/vt/pull/1494#pullrequestreview-871271999

### PR DESCRIPTION
**Pull request by JacobDomagala**
Fixes #1669

### CODE REVIEW

File: src/vt/messaging/active.cc [hunk 1]
```diff
@@ -449,17 +449,7 @@ EventType ActiveMessenger::sendMsgBytes(
...
   }
 
-  if (theContext()->getTask() != nullptr) {
-    auto lb = theContext()->getTask()->get<ctx::LBStats>();
-    if (lb) {
-      auto const already_recorded =
-        envelopeCommStatsRecordedAboveBareHandler(msg->env);
-      if (not already_recorded) {
-        auto dest_elm_id = elm::ElmIDBits::createBareHandler(dest);
-        theContext()->getTask()->send(dest_elm_id, msg_size);
```
Comment (PhilMiller) [inline src/vt/messaging/active.cc:459]: Does `msg_size == base.size()`? We're getting `msg_size` as an argument, so I don't immediately see.

Comment (JacobDomagala) [reply src/vt/messaging/active.cc:459]: I think it's not always the case, I'll fix that. Thanks

File: src/vt/messaging/active.h [hunk 2]
```diff
@@ -1725,6 +1725,18 @@ struct ActiveMessenger : runtime::component::PollableComponent<ActiveMessenger>
...
+  /**
+   * @brief Record LB's statistics for sending a message
+   *
+   * \param[in] dest the destination of the message
+   * \param[in] base the message base pointer
+   * \param[in] msg_size the size of the message being sent
+   */
+  void recordLbStatsCommForSend(
+    NodeType const& dest, MsgSharedPtr<BaseMsgType> const& base,
+    MsgSizeType const& msg_size
```
Comment (jstrzebonski) [inline src/vt/messaging/active.h:1737]: I think that dest and msg_size could be passed by value.

**Review by cz4rs — APPROVED** [review]
Looks good to me :+1:

**Review by lifflander — APPROVED** [review]
Looks good to me!

**Pull request merged**